main.go

```diff
@@ -137,6 +140,12 @@ func coverage(coverpkg, pkg, covermode, cpu, parallel, timeout string, short, v
 	if v {
 		args = append(args, "-v")
 	}
+	if race {
+		if covermode != "" && covermode != "atomic" {
+			log.Fatalf("cannot use race flag and covermode=%s. See more detail on golang/go#12118.", covermode)
```

Comment:
If we use `log.Fatalf` here, `defer`ed function won't be properly called.

I think we have two option.

1. return `err` here and handle `err` from `coverage()` properly.
  - I think I should have handle the error from `coverage()` regardless this pull-request... sorry
2. check these flag more early like the start of `run()` function.

I prefer option 2.

Anyway, I think I should refactor the code... additional test flag won't be changed , so we can move the code from `coverage()` to `run()`.